src/Microsoft.Azure.WebJobs.ServiceBus/Config/ServiceBusExtensionConfig.cs

```diff
@@ -51,6 +52,20 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
+            // Register an exception handler for background exceptions
+            // coming from MessageReceivers.
+            //
+            // The message options is a host level instance that is shared
+            // across all bindings, so we have to subscribe to it at the
+            // host level.
+            Config.MessageOptions.ExceptionReceived += (s, e) =>
+            {
+                if (!e.Exception.IsWrappedExceptionTransient())
+                {
+                    context.Trace.Error($"MessageReceiver error (Action={e.Action})", e.Exception);
```

Comment:
I'm assuming that the TraceWriter is all wired up to AppInsights if AI is enabled @brettsam ?

Reply:
Updated to conditionally log to ILogger as well :)